atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/asts/progress/AbstractDefaultBucketProgressStoreTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.asts.progress;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.impl.KvsManager;
+import com.palantir.atlasdb.schema.TargetedSweepSchema;
+import com.palantir.atlasdb.sweep.asts.ImmutableSweepableBucket;
+import com.palantir.atlasdb.sweep.asts.SweepStateCoordinator.SweepableBucket;
+import com.palantir.atlasdb.sweep.queue.ShardAndStrategy;
+import com.palantir.atlasdb.table.description.Schemas;
+import com.palantir.atlasdb.table.description.SweeperStrategy;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+// This exists as an abstract class, because the semantics of the atomic table operations are tricky, and novel as
+// part of the auto-scaling sweep project.
+public class AbstractDefaultBucketProgressStoreTest {
```

Comment:
not actionable: the name is consistent with AbstractTargetedSweeperTest, but it's mildly confusing since this isn't an abstract class! (At least AbstractTargetedSweeperTest extends an abstract class)

Reply:
Yeah will add a comment at least. I don't really like this naming convention, but yeah it's consistent.

Reply:
actually I think this should be abstract, for what it's worth!